### PR TITLE
Remove "available since" comments

### PR DESCRIPTION
**Pull request by frangio**
The documentation includes comments saying "Available since vX.Y" meant to indicate that a feature was introduced in a minor, because the docs are only versioned by major. We don't need this since we're moving to 5.0.

~Blocked by https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4423 because the changes to `upgradeable.patch` conflict.~

Fixes LIB-948

### CODE REVIEW

File: contracts/token/ERC1155/utils/ERC1155Receiver.sol
Comment (frangio) [inline contracts/token/ERC1155/utils/ERC1155Receiver.sol:14]: I don't really understand why we have separate ERC1155Receiver and ERC1155Holder contracts. I think we should merge them, and only keep ERC1155Holder.

File: scripts/upgradeable/upgradeable.patch
Comment (Amxx) [inline scripts/upgradeable/upgradeable.patch:145]: This part of the patch is removing some natspec comments from the vanilla version.
It looks to me that these natspec comments are only usefull in an upgradeable context. So why do we have them in the vanilla version in the first place ?

Comment (frangio) [reply scripts/upgradeable/upgradeable.patch:145]: The vanilla contract is upgradeable-safe even though it uses immutable variables. If we keep these annotations, the transpiler will keep the immutable variables instead of converting them. That said I think the patch removes all immutable variables anyway.

As a side note, I'm realizing that using `unsafe-allow state-variable-assignment` in this way is bad because we only want it to apply to state-variable-immutable and this will allow assignments to any storage variable.

Comment (Amxx) [reply scripts/upgradeable/upgradeable.patch:145]: > The vanilla contract is upgradeable-safe.

Is that something we want the upgrade plugin to detect ? Using the vanila EIP712 as an implementation for proxy is dangerous.

Comment (frangio) [reply scripts/upgradeable/upgradeable.patch:145]: >Using the vanilla EIP712 as an implementation for proxy is dangerous.

Is it? I think the annotation is there because it is not dangerous.

Comment (Amxx) [reply scripts/upgradeable/upgradeable.patch:145]: You're right. I forgot about the `address(this) == _cachedThis` check. Its indeed safe.

**Review by ernestognw — APPROVED** [review]
There's a missing import to remove from `scripts/generate/templates/SafeCast.js` that might require another commit but LGTM, let me know if you need reapproving

**Review by ernestognw — APPROVED** [review]

**Pull request merged**
